### ISSUE TITLE
nixos-rebuild output within nixos-enter is confusing

### ISSUE THREAD

**Issue by ElvishJerricco** [NixOS/nixpkgs#356432]
When running `nixos-rebuild` in a `nixos-enter`, there are two messages it can output that are really confusing.

1. We try `systemd-run` to check if it's working, so that we can avoid using it if it isn't: https://github.com/NixOS/nixpkgs/blob/511664f887644142fd07a4cb91ac363007690e8c/pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh#L871-L872 But because we're running in a chroot, we get this message:
    ```
    System has not been booted with systemd as init system (PID 1). Can't operate.
    Failed to connect to bus: Host is down
    ```
    And we don't get the log message explaining this because it uses `logVerbose`. Ideally we should not see this error message at all.
2. If a user mistakenly uses `nixos-rebuild switch` or `nixos-rebuild test`, the output is more like this:
    ```
    System has not been booted with systemd as init system (PID 1). Can't operate.
    Failed to connect to bus: Host is down
    Error: Failed to open dbus connection

    Caused by:
        Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
    warning: error(s) occurred while switching to the new configuration
    ```
    Instead, both `nixos-rebuild` and `switch-to-configuration` should be checking if it even makes sense to attempt `switch` or `test`. Upstream systemd recommends checking for the existence of a directory at `/run/systemd/system` to check if we're running with systemd.

/cc @emilazy

**Comment by emilazy** [comment]
cc @jmbaur

**Comment by emilazy** [comment]
cc @thiagokokada for `nixos-rebuild-ng` test cases though :)

**Comment by thiagokokada** [comment]
CC @roberth since this will probably go to the new `apply` script instead of `nixos-rebuild-ng` (the plan for now is to not reimplement the `systemd-run` logic and instead leave it to the `apply` script, but this may change of course).

**Comment by roberth** [comment]
That's right. The new interface between `nixos-rebuild` and `toplevel` is the `apply` script, and it is located such that everything that inherently needs to run on the target host is executed by the `apply` script.
That includes the `systemd-run` wrapping and manipulation of the profile.

I can confirm that stc requires no change for this issue.

**Comment by ElvishJerricco** [comment]
> I can confirm that stc requires no change for this issue.

I still think it would be a good idea for stc to bail if it shouldn't be attempting to switch. It's not uncommon for people to run that script manually.

**Comment by thiagokokada** [comment]
Same as https://github.com/NixOS/nixpkgs/issues/331729, this issue should be fixed in `nixos-rebuild-ng`, that is now the default, so I will close this issue.
